### PR TITLE
feat: [CDS-54247]: Support accessing of connector attributes for Deployment template

### PR DESCRIPTION
**Pull request by alluvamsi-harness**
The connector attributes for Secret Mangers can be accessed using expressions for Deployment template
Example: The one of the attribute of custom secrets manager can be accessed using <+stage.spec.infrastructure.output.variables.CustomAcc.spec.template.versionLabel>
Tested by creating a pipeline with Deployment template and by printing the expression values using shell script
Test Plan: https://docs.google.com/spreadsheets/d/1t0Cy6hRr1ngv6aRp1QurIydOol7bHCU9ANKe1QmbFMw/edit?usp=sharing
image

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/46326)

### CODE REVIEW

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/gcpkmsconnector/outcome/GcpKmsConnectorOutcomeDTO.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  @NotNull
+  @SecretReference
+  SecretRefData credentials;
+  @Schema(description = SecretManagerDescriptionConstants.DEFAULT) private boolean isDefault;
+
+  private boolean harnessManaged;
+  @Schema(description = SecretManagerDescriptionConstants.DELEGATE_SELECTORS) private Set<String> delegateSelectors;
+
+  @Override
+  public List<DecryptableEntity> getDecryptableEntities() {
```
Comment (nishantsaini20) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/gcpkmsconnector/outcome/GcpKmsConnectorOutcomeDTO.java:58]: Same here. Do we need this for any use case?

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/gcpsecretmanager/outcome/GcpSecretManagerConnectorOutcomeDTO.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  @Schema(description = SecretManagerDescriptionConstants.GOOGLE_SECRET_MANAGER_CREDENTIALS)
+  @NotNull
+  @SecretReference
+  SecretRefData credentialsRef;
+
+  @Schema(description = SecretManagerDescriptionConstants.DELEGATE_SELECTORS) Set<String> delegateSelectors;
+
+  @Override
+  public List<DecryptableEntity> getDecryptableEntities() {
```
Comment (nishantsaini20) [inline 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/gcpsecretmanager/outcome/GcpSecretManagerConnectorOutcomeDTO.java:46]: Same here and other similar classes. Do we need this for any use case?

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/awssecretmanager/outcome/AwsSecretManagerOutcomeDTO.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+  @Schema(description = SecretManagerDescriptionConstants.AWS_REGION_SM) @NotNull private String region;
+  @Schema(description = SecretManagerDescriptionConstants.DEFAULT) private boolean isDefault;
+  @Schema(description = SecretManagerDescriptionConstants.HARNESS_MANAGED) @JsonIgnore private boolean harnessManaged;
+
+  @Schema(description = SecretManagerDescriptionConstants.AWS_SECRET_NAME_PREFIX) private String secretNamePrefix;
+  @Schema(description = SecretManagerDescriptionConstants.DELEGATE_SELECTORS) private Set<String> delegateSelectors;
+
+  @Override
+  public List<DecryptableEntity> getDecryptableEntities() {
```
[the remark this answers is not in the document]
Comment (alluvamsi-harness) [reply 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/awssecretmanager/outcome/AwsSecretManagerOutcomeDTO.java]: removed

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/azurekeyvaultconnector/outcome/AzureKeyVaultConnectorOutcomeDTO.java [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  @Schema(description = "This specifies the Azure Environment type, which is AZURE by default.")
+  private AzureEnvironmentType azureEnvironmentType = AZURE;
+  @Schema(description = SecretManagerDescriptionConstants.DELEGATE_SELECTORS) private Set<String> delegateSelectors;
+
+  @Schema(description = "Boolean value to indicate if managed identity is used") private Boolean useManagedIdentity;
+  @Schema(description = "Managed Identity Type") private AzureManagedIdentityType azureManagedIdentityType;
+  @Schema(description = "Client Id of the ManagedIdentity resource") String managedClientId;
+
+  @Override
+  public List<DecryptableEntity> getDecryptableEntities() {
```
[the remark this answers is not in the document]
Comment (alluvamsi-harness) [reply 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/azurekeyvaultconnector/outcome/AzureKeyVaultConnectorOutcomeDTO.java]: removed

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/outcome/CustomSecretManagerConnectorOutcomeDTO.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  @SecretReference
+  @Schema(description = SecretManagerDescriptionConstants.CUSTOM_AUTH_TOKEN)
+  private SecretRefData connectorRef;
+
+  private String host;
+  private String workingDirectory;
+  @NotNull private TemplateLinkConfigForCustomSecretManager template;
+
+  @Override
+  public List<DecryptableEntity> getDecryptableEntities() {
```
[the remark this answers is not in the document]
Comment (alluvamsi-harness) [reply 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/customsecretmanager/outcome/CustomSecretManagerConnectorOutcomeDTO.java]: removed

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/gcpkmsconnector/outcome/GcpKmsConnectorOutcomeDTO.java (hunk 1 above)
Comment (alluvamsi-harness) [reply 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/gcpkmsconnector/outcome/GcpKmsConnectorOutcomeDTO.java]: removed

File: 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/gcpsecretmanager/outcome/GcpSecretManagerConnectorOutcomeDTO.java (hunk 2 above)
Comment (alluvamsi-harness) [reply 954-connector-beans/src/main/java/io/harness/delegate/beans/connector/gcpsecretmanager/outcome/GcpSecretManagerConnectorOutcomeDTO.java]: removed

**Comment by alluvamsi-harness** [comment]
trigger messagecheck

**Comment by alluvamsi-harness** [comment]
trigger ss

**Comment by alluvamsi-harness** [comment]
trigger copyrightcheck

**Review by RishabhGupta34 — APPROVED** [review]

**Review by nishantsaini20 — APPROVED** [review]

**Review by VikasMaddukuriHarness — APPROVED** [review]

**Review by tiwaritejas — APPROVED** [review]
> Reviewed 7 of 18 files at r1, 32 of 32 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/46326)* status: 39 file(s) to review. 6. unresolved discussions.

**Comment by alluvamsi-harness** [comment]
trigger review

**Comment by alluvamsi-harness** [comment]
trigger review

**Pull request merged**